app/api/tasks.py

```diff
@@ -42,10 +49,45 @@ def get_and_check_project(self, request, project_pk, perms = ('view_project', ))
         except ObjectDoesNotExist:
             raise exceptions.NotFound()
         return project
-    
+
+    @detail_route(methods=['post'])
+    def cancel(self, request, pk=None, project_pk=None):
+        self.get_and_check_project(request, project_pk, ('change_project',))
+        try:
+            task = self.queryset.get(pk=pk, project=project_pk)
+        except ObjectDoesNotExist:
+            raise exceptions.NotFound()
+
+        task.pending_action = task.PendingActions.CANCEL
+        task.last_error = None
+        task.save()
+
+        # Call the scheduler (speed things up)
+        scheduler.process_pending_tasks(background=True)
```

Comment:
This is totally not actionable, but I can't shake the feeling that you may be going down a road where you're reinventing celery.